engines/grim/grim.cpp

```diff
@@ -1025,8 +1047,29 @@ void GrimEngine::mainLoop() {
 		while (g_system->getEventManager()->pollEvent(event)) {
 			// Handle any buttons, keys and joystick operations
 			Common::EventType type = event.type;
-			if (type == Common::EVENT_KEYDOWN || type == Common::EVENT_KEYUP) {
+
+			bool doubleClick = false;
+			// parse special gestures
+			static unsigned int _lastClick = 0;
+			if (type == Common::EVENT_LBUTTONDOWN) {
+				unsigned int currentTime = g_system->getMillis();
+				doubleClick = (currentTime - _lastClick) < 500;
+				_lastClick = currentTime;
+			}
+			if (type == Common::EVENT_MOUSEMOVE) {
+				_cursor->updatePosition(event.mouse);
+				_hotspotManager->hover(_cursor->getPosition());
+			} else if (type == Common::EVENT_MBUTTONDOWN ||
+			           (type == Common::EVENT_LBUTTONDOWN && event.kbd.hasFlags(Common::KBD_CTRL))) {
+				Common::KeyState kbd(Common::KEYCODE_i);
```

Comment:
looks like CTRL + LMB is mapped to 'I'. I think we should handle now with generic keys mapper.

Reply:
It opens the inventory.

Reply:
keeping only LMB sounds better

Reply:
Done

Reply:
hmm, actually I wonder if middle button can be mapped in generic way

Reply:
I can't find an example for that. If you have a way, let me know.

Reply:
default engine mapping:
act = new Action(kStandardActionMiddleClick, _("Middle Click"));
act->addDefaultInputMapping("MOUSE_MIDDLE");
act->setMiddleClickEvent();
engineKeyMap->addAction(act);

Grim mapping:
act = new Action("INVT", _("Inventory"));
act->setKeyEvent(KeyState(KEYCODE_d, 'i'));
act->addDefaultInputMapping("JOY_Y");
engineKeyMap->addAction(act);

I don't remember how it works, but maybe try add:
act->setMiddleClickEvent();

or overing engines default

Reply:
I looked at it, but from what I see, `setMiddleClickEvent()` discards the previously set key.

Reply:
two separated mappings doesn't looks good, so ok, keep as it is, but only for middle button

Reply:
Yes, that's already done. Thanks.